great_expectations/expectations/expectation.py

```diff
@@ -717,8 +725,27 @@ def _diagnostic_unexpected_statement_renderer(
         assert result, "Must provide a result object."
         success: Optional[bool] = result.success
         result_dict: dict = result.result
-
-        if result.exception_info["raised_exception"]:
+        exception: dict = {
```

Comment:
Typically I wouldn't add an explicit type annotation here, as it is self-evident.